pubsub_nats_test.go

```diff
@@ -203,50 +203,8 @@ func TestPublish(t *testing.T) {
 			natsOptions:     []NATSOption{},
 		},
 		{
-			description: "should return an error if no NATS client had been connected",
-			publication: NewPublication("test topic"),
-			setup: func(t *testing.T, mockClient *mocks.MockNATSClient, pub *Publication) {
-				mockClient.
-					EXPECT().
-					Publish(pub.Topic, gomock.Any()).
-					// should never be called in this case!
-					Times(0)
-			},
-			expectedErrType: errors.New(""),
-			natsOptions: []NATSOption{
-				// notice how we pass a nil client explicitly to simulate the failure scenario
-				// desired by this test
-				natsOptClient(nil),
-			},
-		},
-		{
-			description: "should return an error if passed in a nil publication",
-			publication: nil,
-			setup: func(t *testing.T, mockClient *mocks.MockNATSClient, pub *Publication) {
-				mockClient.
-					EXPECT().
-					Publish(gomock.Any(), gomock.Any()).
-					// should never be called in this case!
-					Times(0)
-			},
-			expectedErrType: errors.New(""),
-			natsOptions:     []NATSOption{},
-		},
-		{
-			description: "should return an error if Publish returns an error",
-			publication: NewPublication("test topic"),
-			setup: func(t *testing.T, mockClient *mocks.MockNATSClient, pub *Publication) {
-				mockClient.
-					EXPECT().
-					Publish(pub.Topic, gomock.Any()).
-					Return(errors.New("failed to publish")).
-					Times(1)
-			},
-			expectedErrType: errors.New(""),
-			natsOptions:     []NATSOption{},
-		},
-		{
-			description: "should be able to provide a custom reply validator using the NATSOptPublishRequireAck option",
+			description: "should be able to request to receive both an ACK and a response to my publication using " +
+				"NATSOptPublishRequireAck & NATSOptRespondToChannel options",
```

Comment:
This is a new test, not sure why the diff is so confusing. Local diff was much cleaner...thanks :octocat: 😡